src/RestaurantDetailPage.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { useParams } from 'react-router-dom';
+
+import { loadRestaurant } from './actions';
+
+import RestaurantDetailContainer from './RestaurantDetailContainer';
+
+export default function RestaurantDetailPage() {
+  const { restaurantId } = useParams();
```

Comment:
```suggestion
export default function RestaurantDetailPage({ params }) {
  const { restaurantId } = params || useParams();
```

위와 같이 고친다면 어떨까요? 테스트에서 어떻게 변경하면 useParams를 가짜로 구현하지 않아도 될까요?

Reply:
..?! 이 부분은 제가 학습이 부족했던 것 같습니다. 
저는 라우팅으로 들어오는 경로를 받으려면 무조건 useParams를 써야되는 줄 알았습니다. 좀 더 학습하고 개선해보겠습니다.

Reply:
죄송합니다 제가 완전히 잘못 이해했었네요 ㅎㅎ;;
수형님 말씀주신 부분은 props가 없을 때(실제 동작할 때)는 useParams를 사용하고,
테스트에서는 해당 컴포넌트에 제가 원하는 값(useParams로 전달되기를 바라는 값)을 받아서 사용하도록 개선하는 방향을 제시해주신 것 같은데 의도하신 바가 이게 맞나요 ?_?
추가로 궁금한 점이 이런 식으로 구현 코드에 테스트를 위한 부분(실제 동작에는 영향이 없지만)을 넣어도 상관 없을까요? 유지보수할 때 저 외의 다른 사람이 보면 '엥 이건 왜 있지?' 싶을 수도 있을 거 같아서요. 수형님께서는 평소 이런 경우 어떻게 처리하시는지 궁금합니다!

Reply:
전 별로 상관 없다고 생각해요. 인터페이스는 다양해도 좋다고 봅니다.

Java에서도 테스트를 위한 생성자도 허용하는 걸요.